enterprise/glassfish.common/src/org/netbeans/modules/glassfish/common/GlassfishInstanceProvider.java

```diff
@@ -251,21 +260,21 @@ public GlassfishInstance getGlassfishInstance(String uri) {
      * @param si GlassFish server instance to be added.
      */
     public void addServerInstance(GlassfishInstance si) {
-        synchronized(instanceMap) {
-            try {
-                instanceMap.put(si.getDeployerUri(), si);
-                activeDisplayNames.add(si.getDisplayName());
-                if (instanceMap.size() == 1) { // only need to do if this first of this type
-                    RegisteredDDCatalog catalog = getDDCatalog();
-                    if (null != catalog) {
-                        catalog.refreshRunTimeDDCatalog(this, si.getGlassfishRoot());
-                    }
+        try {
```

Comment:
The code makes assumptions about sychronisation (see line 268 in new code, line 258 in old code). As the synchronisation was removed, between line 264 (new code) and line 268 (new code) the map might be modified and the the following code is not executed.